test/meta_transaction_test.ts

```diff
@@ -0,0 +1,742 @@
+            },
+            body: {
+                code: GeneralErrorCodes.ValidationError,
+                reason: 'Validation Failed',
+                validationErrors: [
+                    {
+                        field: 'buyToken',
+                        // TODO(jalextowle): This seems like the wrong error message.
+                        code: ValidationErrorCodes.ValueOutOfRange,
+                        reason: 'Could not find token `INVALID`',
+                    },
+                ],
+            },
+            takerAddress: true,
+        },
+        {
+            description: 'Invalid `sellToken`',
+            queryParams: {
+                ...DEFAULT_QUERY_PARAMS,
+                sellToken: 'INVALID',
+            },
+            body: {
+                code: GeneralErrorCodes.ValidationError,
+                reason: 'Validation Failed',
+                validationErrors: [
+                    {
+                        field: 'sellToken',
+                        // TODO(jalextowle): This seems like the wrong error message.
+                        code: ValidationErrorCodes.ValueOutOfRange,
+                        reason: 'Could not find token `INVALID`',
+                    },
+                ],
+            },
+            takerAddress: true,
+        },
+        {
+            description: 'Insufficient Liquidity',
+            queryParams: DEFAULT_QUERY_PARAMS,
+            body: {
+                code: GeneralErrorCodes.ValidationError,
+                reason: 'Validation Failed',
+                validationErrors: [
+                    {
+                        code: ValidationErrorCodes.ValueOutOfRange,
+                        field: 'buyAmount',
+                        reason: 'INSUFFICIENT_ASSET_LIQUIDITY',
+                    },
+                ],
+            },
+            takerAddress: true,
+        },
+    ];
+
+    describe('/price tests', () => {
+        context('failure tests', () => {
+            for (const testCase of testCases) {
+                it(`${testCase.description}`, async () => {
+                    await assertFailureAsync(`${META_TRANSACTION_PATH}/price`, testCase);
+                });
+            }
+        });
+
+        context('success tests', () => {
+            let meshUtils: MeshTestUtils;
+
+            beforeEach(async () => {
+                await blockchainLifecycle.startAsync();
+                await setupMeshAsync(SUITE_NAME);
+                meshUtils = new MeshTestUtils(provider);
+                await meshUtils.setupUtilsAsync();
+            });
+
+            afterEach(async () => {
+                await blockchainLifecycle.revertAsync();
+                await teardownMeshAsync(SUITE_NAME);
+            });
+
+            // NOTE(jalextowle): Spin up a new Mesh instance so that it will
+            // be available for future test suites.
+            after(async () => {
+                await setupMeshAsync(SUITE_NAME);
```

Comment:
why do we setup mesh again, when every suite has a `beforeEach` that sets up mesh?